File: test-suite/src/test/java/example/micronaut/aiservice/tools/OllamaToolTest.java
package example.micronaut.aiservice.tools;

import io.micronaut.context.annotation.Property;
import io.micronaut.langchain4j.testutils.OllamaTestPropertyProvider;
import io.micronaut.langchain4j.testutils.OllamaUtils;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertTrue;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInstance;
import org.junit.jupiter.api.condition.DisabledIfEnvironmentVariable;
import org.testcontainers.junit.jupiter.Testcontainers;

@Property(name = "langchain4j.ollama.model-name", value = OllamaUtils.TOOL_MODEL_NAME)
@Testcontainers(disabledWithoutDocker = true)
@MicronautTest(startApplication = false)
@TestInstance(TestInstance.Lifecycle.PER_CLASS)
@DisabledIfEnvironmentVariable(named = "CI", matches = ".*")
class OllamaToolTest implements OllamaTestPropertyProvider {
    @Test
    void testInjectTools(CompanyBot bot) {
        assertNotNull(bot);
        String response = bot.ask("When was the PRIVACY document updated?");
        assertTrue(response.contains("2013"), response);
    }
}

File: test-suite-python/src/test/python/example/micronaut/aiservice/tools/OllamaToolTest.py
from typing import Annotated

from jakarta.inject import Inject
from micronaut.context.annotation import Property
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test

from example.micronaut.aiservice.tools.CompanyBot import CompanyBot


@Property(name="langchain4j.ollama.model-name", value="qwen2.5:0.5b")
@MicronautTest(startApplication=False, environments=["ollama"])
class OllamaToolTest:
    bot: Annotated[CompanyBot, Inject]

    @Test
    def test_inject_tools(self):
        assert self.bot is not None
        response = self.bot.ask("When was the PRIVACY document updated?")
        assert "2013" in response, response
